### PR TITLE
[Bug]: Results miss data seems that plan pushing down filters incorrectly

### PR DESCRIPTION
**Pull request by badboynt1**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #13871

## What this PR does / why we need it:
fix a bug that order by limit push down incorrectly for limit with offset

### CODE REVIEW

**Comment by matrix-meow** [comment]
@badboynt1 Thanks for your contributions!

Title: [Bug]: Results miss data seems that plan pushing down filters incorrectly

Body: This pull request fixes a bug where the order by limit push down is incorrect for limit with offset.

Changes:
- In the `opt_misc.go` file, a new condition is added to check if the node has an offset. If it does, the offset and limit values are extracted and the limit value is updated to be the sum of the offset and limit. The offset is then set to nil.
- In the `pushdown.result` file, new queries are added to test the fix for the bug. The expected query plans and results are provided.
- In the `pushdown.test` file, new queries are added to test the fix for the bug. The expected query plans and results are provided.

Overall, the changes made in this pull request address the bug related to the incorrect order by limit push down for limit with offset. The code changes correctly update the limit value and the test cases verify the fix.

However, there are a few issues and suggestions for improvement:

1. **Code Optimization**: The code changes could be optimized by avoiding unnecessary variable assignments and imports. For example, in the `opt_misc.go` file, the `x` variable is assigned the value of `lit.GetValue().(*plan.Literal_I64Val)` but is not used anywhere. This assignment can be removed to improve code readability.

2. **Error Handling**: The code changes do not handle any potential errors that may occur during the extraction of offset and limit values. It is important to handle errors properly to prevent unexpected behavior or crashes. Consider adding error handling code to handle any potential errors that may occur during the extraction process.

3. **Security Issue**: The code changes do not introduce any security issues.

In summary, the changes made in this pull request address the bug related to the incorrect order by limit push down for limit with offset. However, there are opportunities for code optimization and error handling improvements. No security issues were identified.

**Review by heni02 — APPROVED** [review]

File: pkg/sql/plan/opt_misc.go [hunk 1]
```diff
@@ -1048,6 +1048,15 @@ func (builder *QueryBuilder) pushTopDownToLeftJoin(nodeID int32) {
 	}
 
 	nodePushDown = DeepCopyNode(node)
+
+	if nodePushDown.Offset != nil {
+		offset := nodePushDown.Offset.GetLit().GetI64Val()
```
Comment (aunjgr) [inline pkg/sql/plan/opt_misc.go:1053]: You should handle parameters in PREPARE stmt here

**Review by aunjgr — CHANGES_REQUESTED** [review]

File: pkg/sql/plan/opt_misc.go (hunk 1 above)
Comment (badboynt1) [reply pkg/sql/plan/opt_misc.go]: fixed

**Review by m-schen — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]
